lib/core/util.js

```diff
@@ -614,20 +727,35 @@ function parseRangeHeader (range) {
     : null
 }
 
+/**
+ * @param {Record<string|symbol, any>} obj
+ * @param {string} name
+ * @param {Function} listener
+ */
 function addListener (obj, name, listener) {
   const listeners = (obj[kListeners] ??= [])
   listeners.push([name, listener])
   obj.on(name, listener)
   return obj
 }
 
+/**
+ * @param {Record<string|symbol, any>} obj
+ */
 function removeAllListeners (obj) {
-  for (const [name, listener] of obj[kListeners] ?? []) {
-    obj.removeListener(name, listener)
+  if (obj[kListeners] != null) {
```

Comment:
if obj[kListeners] is not null or undefined, then we know it is an array, so only do the steps if we have somthing set.